src/lib/replication/service/raft_repl_service.cpp

```diff
@@ -175,6 +182,46 @@ void RaftReplService::stop() {
     hs()->logstore_service().stop();
 }
 
+void RaftReplService::monitor_cert_changes() {
+    auto fw = ioenvironment.get_file_watcher();
+    auto cert_change_cb = [this](const std::string filepath, const bool deleted) {
+        LOGINFO("file change event for {}, deleted? {}", filepath, deleted)
+        // do not block file_watcher thread
+        std::thread restart_svc(&RaftReplService::restart_raft_svc, this, filepath, deleted);
+        restart_svc.detach();
+    };
+
+    //monitor ssl cert file
+    if (!fw->register_listener(ioenvironment.get_ssl_cert(), "hs_ssl_cert_watcher", cert_change_cb)) {
+        LOGERROR("Failed to register listner, {} to watch file {}, Not monitoring cert files",
+                   "hs_ssl_cert_watcher", ioenvironment.get_ssl_cert());
+    }
+    //monitor ssl key file
+    if (!fw->register_listener(ioenvironment.get_ssl_key(), "hs_ssl_key_watcher", cert_change_cb)) {
+        LOGERROR("Failed to register listner, {} to watch file {}, Not monitoring cert files",
+                   "hs_ssl_key_watcher", ioenvironment.get_ssl_key());
+    }
+}
+
+void RaftReplService::restart_raft_svc(const std::string filepath, const bool deleted){
+    if (deleted && !wait_for_cert(filepath)) {
+        LOGINFO("file {} deleted, ", filepath)
+        // wait for the deleted file to be added again
+        throw std::runtime_error(fmt::format("file {} not found! Can not start grpc server", filepath));
+    }
+    const std::unique_lock lock(raft_restart_mutex);
+    m_msg_mgr->restart_server();
+    if (deleted) { monitor_cert_changes(); }
+}
+
+bool RaftReplService::wait_for_cert(const std::string& filepath) {
+    for (auto i = cert_change_timeout; i > 0; --i) {
```

Comment:
nit: cert_change_timeout looks like an absolute time ,  but the implementation make it to `cert_change_timeout * cert_check_sleep`

Suggesting make `cert_change_timeout` an absolute timeout by `i = cert_change_timeout/cert_check_sleep`

Reply:
Sure.